autobot/telegram/parser.py

```diff
@@ -238,16 +238,459 @@ def _entity_parser (entity:dict):
         return msg_obj
 
     # methods to parse games objects
+    def _parse_game (self,key:str,val:dict) -> Game|GameHighScore:
+        game_obj = None
+
+        if (key == "game"):
+            game_obj = Game()
+            for k,v in val.items():
+                match k:
+                    case "photo":
+                        pht = self._parse_photosize(k,v)
+                        setattr(game_obj,k,pht)
+                    case "animation":
+                        anim = self._parse_animation(k,v)
+                        setattr(game_obj,k,anim)
+                    case _:
+                        setattr(game_obj,k,v)
+
+        elif (key == "game_high_score"):
+            game_obj = GameHighScore()
+            for k,v in val.items():
+                setattr(game_obj,k,v)
+
+        return game_obj
     
     
     # methods to parse inline objects
+    def _parse_inlinequeryresults (self,key:str,val:dict) -> InlineQueryResult|InlineQueryResultArticle|InlineQueryResultAudio|InlineQueryResultCachedAudio|InlineQueryResultCachedDocument|InlineQueryResultCachedGif|InlineQueryResultCachedMpeg4Gif|InlineQueryResultCachedPhoto|InlineQueryResultCachedSticker|InlineQueryResultCachedVideo|InlineQueryResultCachedVoice|InlineQueryResultContact|InlineQueryResultDocument|InlineQueryResultGame|InlineQueryResultGif|InlineQueryResultLocation|InlineQueryResultMpeg4Gif|InlineQueryResultPhoto|InlineQueryResultVenue|InlineQueryResultVideo|InlineQueryResultVoice:
```

Comment:
The return types here look overwhelming to the eyes. I suggest you use type aliasing, something like `InlineQueryResults = list[all_inlinequeryresultobjects]` so that you can do something like. You can declare these types at the beginning of the file just after all the imports
`def _parse_inlinequeryresult (self,key:str,val:dict) -> InlineQueryResults:`

Reply:
Something like this ?

```python
InlineQueryResults = list[InlineQueryResult|InlineQueryResultArticle|InlineQueryResultAudio|InlineQueryResultCachedAudio|InlineQueryResultCachedDocument|InlineQueryResultCachedGif|InlineQueryResultCachedMpeg4Gif|InlineQueryResultCachedPhoto|InlineQueryResultCachedSticker|InlineQueryResultCachedVideo|InlineQueryResultCachedVoice|InlineQueryResultContact|InlineQueryResultDocument|InlineQueryResultGame|InlineQueryResultGif|InlineQueryResultLocation|InlineQueryResultMpeg4Gif|InlineQueryResultPhoto|InlineQueryResultVenue|InlineQueryResultVideo|InlineQueryResultVoice]
```

Reply:
You have to replace the vertical bars with commas.